lib/AST/SubstitutionMap.cpp

```diff
@@ -395,6 +398,40 @@ SubstitutionMap::combineSubstitutionMaps(const SubstitutionMap &firstSubMap,
     });
 }
 
+void SubstitutionMap::verify() const {
+#ifndef NDEBUG
```

Comment:
@jckarter I agree with you. But introducing an -enable-verifiers flag should be a separate commit, which would introduce a flag and update all the places where we currently compile out the code in release builds.